api/src/main/java/org/openmrs/Obs.java

```diff
@@ -1080,11 +1080,31 @@ public String getValueAsString(Locale locale) {
 		return "";
 	}
 	
-	private static DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd");
+	private static ThreadLocal<DateFormat> dateFormat = new ThreadLocal<DateFormat>() { // Made DateFormat Threadsafe using ThreadLocal		

```

Comment:
Many google results e.g. http://stackoverflow.com/a/6840897 say ThreadLocal is the fastest option, but I haven't seen exact numbers quoted.